client/app/containers/index.js

```diff
@@ -83,17 +84,31 @@ class App extends React.Component {
     return api.getTree(newPath).then(mapToState).catch(this.handleError);
   }
 
-  componentDidMount() {
-    const newPath = window.location.search || this.state.path;
+  createBreadcrumbs(path) {
+    let breadcrumbs = path.split('/');
 
-    this.formNewState(newPath).then(newState => {
-      window.history.replaceState(newState, null, newPath);
-      window.onpopstate = ({ state }) => this.setState(state);
-    }).catch(this.handleError);
+    breadcrumbs.splice(0, 1, '.');
```

Comment:
why not have path passed from the top be already just path and not something with `?path=`?